components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -513,6 +540,25 @@ private Application createOAuthApplication(ApplicationRegistrationRequest regist
         return application;
     }
 
+    /**
+     * Check whether the software statement is mandatory and the software statement is empty.
+     *
+     * @param softwareStatement Software statement.
+     * @return True if the software statement is mandatory and the software statement is empty.
+     * @throws DCRMServerException If an error occurred while checking the software statement.
+     */
+    public boolean isSSAMandatedAndSSAisEmpty(String softwareStatement)
```

Comment:
We should not do multiple things from the same method. Here you do two 2 things in the same method and the method name denotes that too as you have used 'And'. Let's use 2 private methods to check  isSSAMandated , isSSAEmpty

Reply:
Fixed with [bcbf429](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2344/commits/bcbf429f60dda15052023305a458026998d9afa3)